src/v/cluster/data_migration_worker.cc

```diff
@@ -187,37 +201,48 @@ ss::future<errc> worker::do_work(managed_ntp_cit it) noexcept {
 ss::future<errc> worker::do_work(
   const model::ntp& ntp,
   state sought_state,
-  const inbound_partition_work_info& pwi) {
+  const inbound_partition_work_info&) {
     vassert(
-      sought_state == state::prepared,
+      false,
```

Comment:
why do we always assert here ?

Reply:
Because no partition-level work is needed for inbound migrations: https://github.com/redpanda-data/redpanda/pull/22881/files/cd54b23b22027fd96474145b3b19e8bdc7db7148#diff-8cee819c5afa18f1d919160dea122ff31f5ecd7710e3702e07787b61b3c4b403L1058-L1062